src/app/_hooks/useFilter.ts

```diff
@@ -0,0 +1,39 @@
+import { useCallback, useMemo } from 'react'
+
+import { type NextServerSearchParams } from '@/types/searchParams'
+import { type Object } from '@/types/utils'
+
+import { ALL_FILTERS_ID } from '@/constants/filterConstants'
+
+import { normalizeQueryParam } from '@/utils/queryUtils'
+
+export type UseFilterProps<Entry extends Object> = {
+  searchParams: NextServerSearchParams
+  entries: Array<Entry>
+  filterKey: string
+  filterFn: (entry: Entry, query?: string) => boolean
+}
+
+export function useFilter<Entry extends Object>({
+  searchParams,
+  entries,
+  filterKey,
+  filterFn,
```

Comment:
In this PR, `filterKey` and `filterFn` are the same in both use cases. Will this change?

Reply:
Yes, it will change when we add location to the mix.

```ts
const { filteredResults } = useFilter({
  searchParams,
  entries: sortedResults,
  filterKey: LOCATION_KEY,
  filterFn: filterByLocation,
})
```

The goal of this PR is to make `useFilter` more agnostic and allow filtering by any field.